dotnet/src/Connectors/Connectors.AI.OpenAI/AzureSdk/ClientBase.cs

```diff
@@ -40,7 +40,7 @@ public abstract class ClientBase
         CompleteRequestSettings requestSettings,
         CancellationToken cancellationToken = default)
     {
-        Verify.NotNull(requestSettings, "Completion settings cannot be empty");
+        Verify.NotNull(requestSettings);
```

Comment:
Wouldn't you agree that custom error messages would help developers understanding the context, e.g. as opposed to generic exceptions and line numbers?

Reply:
My changes include using CallerArgumentExpression with NotNull helper. The exception message in this case will be:
```
Value cannot be null. (Parameter 'requestSettings')
```
I think that's more valuable, telling the developer exactly which parameter is the problem. With the custom message:
```
Completion settings cannot be empty
```
the developer needs to know that a) empty means null and b) "completion settings" is referring to the requestSettings parameter.